### PR TITLE
SplitKB Aurora Lily58 port

### PR DESCRIPTION
**Pull request by byte4c**

### CODE REVIEW

File: boards/splitkb/aurora/lily58/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- Support for a power switch for wireless controllers;
+- Support for the splitkb.com tenting puck.
+
+# What is the Aurora Series?
+
+The Aurora series were designed as a homage to open source keyboard kits that have carved their space into the community. Kits that provided something new and unique, kits that resonated with the community and that have both received and have given a lot of love in return.
+
+With the Aurora series, we aim to make these kits available to the community in a way that both honors the original, while also offering the best possible feature set and a consistent, well-documented build experience.
+
+For each kit sold, we give back to the community by donating €1 across [the firmware projects we support](https://opencollective.com/splitkb). This way we support open source projects and benefit everyone using keyboards, whether they use our kits or their own.
```
Comment (kdb424) [inline boards/splitkb/aurora/lily58/README.md:31]: Kinda odd to have this link when KMK isn't listed there.

File: kmk/quickpin/pro_micro/liatris.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import board
```
Comment (kdb424) [inline kmk/quickpin/pro_micro/liatris.py:1]: This entire file seems to be out of place in this PR. May be worth opening a separate PR for this.

Comment (byte4c) [reply kmk/quickpin/pro_micro/liatris.py:1]: The Liatris controller is a controller designed by splitkb.com so it kind of makes sence to have their keyboard and the quickpin for their controller together. But I could also understand having them separate.
You pose the question 'it may be worth'. If you or someone says 'it should be', then I'll be happy to make a separate PR.

File: boards/splitkb/aurora/lily58/README.md (hunk 1 above)
Comment (byte4c) [reply boards/splitkb/aurora/lily58/README.md:31]: This is just a copy of the Readme from the QMK or VIAL repo I think. They don't just sell the keyboards with firmware made by the community, they also actually do make financial contributions (and I'm assuming code contributions). It speaks to their motivation. But there are only so many firmwares they can contribute to themselves, hence this community contribution. If KMK is made compatible using this, it might be the first step to have KMK on that list?

I don't have a horse in this race. From my perspective it's factually correct information. But if it needs to be removed, I'll do just that.

File: boards/splitkb/aurora/lily58/main.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+from kb import KMKKeyboard
+from kmk.modules.split import Split
+from kmk.modules.layers import Layers
+from kmk.modules.holdtap import HoldTap
+from kmk.extensions.rgb import RGB
+from kmk.extensions.peg_oled_display import Oled,OledDisplayMode,OledReactionType,OledData
```
Comment (xs5871) [inline boards/splitkb/aurora/lily58/main.py:6]: The use `peg_oled_display` is discouraged because of QOC, bad performance, not actively maintained.
We recommend to use the `display` extension instead.

File: kmk/quickpin/pro_micro/liatris.py (hunk 2 above)
Comment (xs5871) [reply kmk/quickpin/pro_micro/liatris.py:1]: I generally agree with splitting contributions into "atomic" commits. A seperate commit in the same PR would be nicer, but in this specific case I don't care too much either way.

File: boards/splitkb/aurora/lily58/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+![Aurora Lily58](https://i.imgur.com/PNRXfCUh.jpg)
+
+# Aurora Lily58
+The Aurora Lily58 is a redesign by splitkb.com of the popular Lily58 by Naoki Katahira.
+
+Keyboard Maintainer: [splitkb.com](https://github.com/splitkb)  
+Hardware Supported: Pro Micro 5V/16MHz and compatible.  
+Hardware Availability: [splitkb.com](https://splitkb.com).
```
Comment (xs5871) [inline boards/splitkb/aurora/lily58/README.md:8]:
```suggestion
```
We don't do vendor promotion anymore.

**Review by xs5871 — COMMENTED** [review]
please run `make fix-isort fix-formatting test` for linting

File: boards/splitkb/aurora/lily58/README.md (hunk 1 above)
Comment (xs5871) [reply boards/splitkb/aurora/lily58/README.md:31]: The "we" in that section is definitely misleading. Also, if you copy something verbatim, you have to cite the source.

Comment (byte4c) [reply boards/splitkb/aurora/lily58/README.md]: I've rewritten parts of the readme to have it not be a verbatim copy anymore (apart from the board features). Doing so I hope I tackled most or all of the remarks given about the readme.

File: boards/splitkb/aurora/lily58/main.py (hunk 3 above)
Comment (byte4c) [reply boards/splitkb/aurora/lily58/main.py]: replaced the `peg_oled_display` with the `display` extension

**Comment by byte4c** [comment]
All comments and remarks are fixed. I don't know if i should resolve all those conversations?

**Review by xs5871 — APPROVED** [review]
> All comments and remarks are fixed. I don't know if i should resolve all those conversations?

No, that's alright.
